libfive/src/render/brep/mesh.cpp

```diff
@@ -116,6 +126,115 @@ std::unique_ptr<Mesh> Mesh::render(
                 });
         t.reset(settings);
     }
+    else if (settings.alg == SIMPLEX_DC) {
+        if (settings.progress_handler) {
+            // Pool::build, Edges, SubConsolidator Collapser 2 and 3, 
+            // Intersecter, Vertexer, Mesher, t.reset
+            settings.progress_handler->start({ 1, 1, 1, 1, 1, 1, 1, 1, 1 });
+        }
+        auto t = SimplexDCWorkerPool<3>::build(es, r, settings);
```

Comment:
I cancelled the task pretty quickly after starting it, and somehow one thread seemed to get stuck in ::build, inside ~Root() -> reset(). The task seems cancelled (progressbar is gone), but ntop behaves pretty sluggish. upon attaching the debugger, I'm now finding a single thread waiting (seemingly forever) for some futures, see screenshot below.
![devenv_2020-05-27_20-02-49](https://user-images.githubusercontent.com/59439122/83056810-4eece100-a056-11ea-9fba-f1a6f96c7f5e.png)